spec/jobs/deposit_work_job_spec.rb

```diff
@@ -189,4 +189,28 @@
       expect(Sdr::Repository).not_to have_received(:accession)
     end
   end
+
+  context 'when review rejected but a manager or reviewer deposits' do
+    let(:druid) { druid_fixture }
+    let(:cocina_object) { dro_with_metadata_fixture }
+    let(:work_form) { WorkForm.new(druid:, content_id: content.id) }
+    let(:work) { create(:work, :rejected_review, druid:, collection:) }
+    let(:version_status) { build(:version_status) }
+    let(:collection) { create(:collection, druid: collection_druid_fixture, reviewers: [current_user]) }
+
+    before do
+      allow(Sdr::Repository).to receive(:status).and_return(version_status)
+      allow(ToCocina::Work::Mapper).to receive(:call).and_return(cocina_object)
+      allow(RoundtripSupport).to receive(:changed?).and_return(false)
+    end
+
+    it 'approves the work and deposits' do
+      expect(work.rejected_review?).to be true
+      work.approve!
```

Comment:
Why is `approve!` here?

Reply:
Ah, I could remove this from the test I suppose as it's just a bit extra. Since the approve is in the controller and not the job, I put it in here to verify that it happened without the transition error, but maybe including it in the test is a bit confusing?

Reply:
If the change is in the controller, then this test in of the DepositJob doesn't actually test the fix.

Reply:
Should I do a system test for this small change? I'm not sure I entirely agree that this isn't sufficient as it verifies the approval failure doesn't fail after the job, which is what is happening in the controller. But I'm open to suggestion.

Reply:
Yes, this needs a small system test. Otherwise, it isn't clear that this solved the problem. (That is, the state of work is changed so that it is in a depositable state when passed to the DepositJob.)

The other alternative is to consider moving the change to the DepositJob where it can be tested.

Reply:
And I'd suggest that it is better to have this sort of logic in the job than the controller.

Reply:
I think your `if` would cover that.